exampleCourse/questions/element/dataframe/server.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import prairielearn as pl
+import pandas as pd
+
+
+def generate(data):
+    df = pd.io.parsers.read_csv("breast-cancer-train.dat", header=None)
+
+    df2 = pd.DataFrame(
+        [
+            {
+                "city": "Champaign",
+                "job": "Professor",
+                "age": 35,
+                "time": pd.to_datetime("2022-10-06 12:00"),
+            },
+            {
+                "city": "Sunnyvale",
+                "job": "Driver",
+                "age": 20,
+                "time": pd.to_datetime("2020-05-09 12:00"),
+            },
+            {
+                "city": "Mountain View",
+                "job": "Data Scientist",
+                "age": np.nan,
+                "time": pd.to_datetime("2021-12-14 12:00"),
+            },
+        ]
+    )
+
+    dft = pd.DataFrame(
+        {
+            # Scalars
+            "integer": 1,
+            "numeric": 3.14,
+            "logical": False,
+            "character": "foo",
+            # TODO adding in complex numbers won't deserialize correctly, fix this (somehow?)
```

Comment:
👀 do you plan to address this TODO in this PR?

Reply:
Unfortunately, this probably can't be resolved in this PR (see https://github.com/PrairieLearn/PrairieLearn/pull/6800#discussion_r1070474330)

Reply:
As @eliotwrobson mentioned, this is an upstream pandas bug with the JSON format, c.f.  

https://github.com/pandas-dev/pandas/issues/50782

Reply:
There was some movement in: https://github.com/pandas-dev/pandas/issues/50782

Not sure if the person responding, @dicristina, is an official member of pandas core; but, they confirmed that the `"table"` orientation being employed is missing handling code for complex numbers.
  
> Unlike in the previous case in the `orient="table"` case we have the data type of each column so in theory we should be able to reconstruct the values without doing any work at all. The problem here is that there is no special handling for complex numbers and when the values are read they are passed to the `float` function. The relevant code is in `pandas/io/json/_table_schema.py`.

Reply:
Nice! Seems like it would be relatively straightforward to add a `"complex"` type to go alongside `integer` and `number`.